### PR TITLE
Completed hibernate04

### PR DESCRIPTION
**Pull request by Vedroid**

### CODE REVIEW

File: src/main/java/ua/vedroid/cinema/Main.java [hunk 1]
```diff
@@ -1,18 +1,109 @@
...
+
+        CinemaHall hallLux = new CinemaHall();
+        hallLux.setCapacity(24);
+        hallLux.setDescription("LUX");
+
+        CinemaHall hall = new CinemaHall();
+        hall.setCapacity(80);
+        hall.setDescription("3D");
+
+        //MovieSession
```
Comment (Artificial-Friend) [inline src/main/java/ua/vedroid/cinema/Main.java:44]: mentors are rejecting ANY comments in code and there are a lot of empty lines

**Review by Artificial-Friend — APPROVED** [review]

File: src/main/java/ua/vedroid/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import org.hibernate.query.Query;
+import ua.vedroid.cinema.dao.CinemaHallDao;
+import ua.vedroid.cinema.exception.DataProcessingException;
+import ua.vedroid.cinema.lib.Dao;
+import ua.vedroid.cinema.model.CinemaHall;
+import ua.vedroid.cinema.util.HibernateUtil;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    static final Logger log = LogManager.getLogger(CinemaHallDaoImpl.class);
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/dao/impl/CinemaHallDaoImpl.java]: what is missing here?

File: src/main/java/ua/vedroid/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            session.save(cinemaHall);
+            transaction.commit();
+            log.info("Added new cinemaHall: " + cinemaHall);
+            return cinemaHall;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            String msg = "Can`t insert cinemaHall entity " + cinemaHall;
+            log.error(msg, e);
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/dao/impl/CinemaHallDaoImpl.java]: there is no reason in logging errors at the same time when you are throwing exception - let's remove this log from all catch blocks

File: src/main/java/ua/vedroid/cinema/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -12,6 +14,8 @@
 
 @Dao
 public class MovieDaoImpl implements MovieDao {
+    static final Logger log = LogManager.getLogger(MovieDaoImpl.class);
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/dao/impl/MovieDaoImpl.java]: what is missing?

File: src/main/java/ua/vedroid/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import org.hibernate.query.Query;
+import ua.vedroid.cinema.dao.MovieSessionDao;
+import ua.vedroid.cinema.exception.DataProcessingException;
+import ua.vedroid.cinema.lib.Dao;
+import ua.vedroid.cinema.model.MovieSession;
+import ua.vedroid.cinema.util.HibernateUtil;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    static final Logger log = LogManager.getLogger(MovieSessionDaoImpl.class);
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/dao/impl/MovieSessionDaoImpl.java]: same

File: src/main/java/ua/vedroid/cinema/model/MovieSession.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+@Entity
+@Table(name = "movie_session")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Movie movie;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private CinemaHall cinemaHall;
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/model/MovieSession.java]: let's use `@JoinColumn` here to guarantee the `proper_naming` for our column

File: src/main/java/ua/vedroid/cinema/Main.java [hunk 7]
```diff
@@ -1,18 +1,109 @@
...
+import ua.vedroid.cinema.model.CinemaHall;
 import ua.vedroid.cinema.model.Movie;
+import ua.vedroid.cinema.model.MovieSession;
+import ua.vedroid.cinema.service.CinemaHallService;
 import ua.vedroid.cinema.service.MovieService;
+import ua.vedroid.cinema.service.MovieSessionService;
 
 public class Main {
     private static final Injector injector = Injector.getInstance("ua.vedroid.cinema");
+    private static final Logger log = LogManager.getLogger();
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/Main.java]:
```suggestion
    private static final Logger log = LogManager.getLogger(Main.class);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/ua/vedroid/cinema/model/MovieSession.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+@Entity
+@Table(name = "movie_session")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Movie movie;
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "cinema_hall")
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/model/MovieSession.java]:
```suggestion
    @JoinColumn(name = "cinema_hall_id")
```

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**